src/Cake.Common/Tools/OctopusDeploy/OctopusDeployArgumentBuilder.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Cake.Core;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.OctopusDeploy
+{
+    internal abstract class OctopusDeployArgumentBuilder<T> where T : OctopusDeploySettings
+    {
+        protected readonly ICakeEnvironment Environment;
+        protected readonly ProcessArgumentBuilder Builder;
+        protected readonly T Settings;
+
+        protected OctopusDeployArgumentBuilder(ICakeEnvironment environment, T settings)
+        {
+            Settings = settings;
+            Environment = environment;
+            Builder = new ProcessArgumentBuilder();
+        }
+
+        protected void AppendArgumentIfNotNull(string argumentName, string value)
+        {
+            if (value != null)
+            {
+                Builder.Append("--" + argumentName);
+                Builder.AppendQuoted(value);
+            }
+        }
+
+        protected void AppendArgumentIfNotNull(string argumentName, FilePath value)
+        {
+            if (value != null)
+            {
+                Builder.Append("--" + argumentName);
+                Builder.AppendQuoted(value.MakeAbsolute(Environment).FullPath);
+            }
+        }
+
+        protected void AppendCommonArguments()
+        {
+            Builder.Append("--server");
+            Builder.Append(Settings.Server);
+
+            Builder.Append("--apiKey");
+            Builder.AppendSecret(Settings.ApiKey);
+
+            AppendArgumentIfNotNull("user", Settings.Username);
+
+            if (Settings.Password != null)
+            {
+                Builder.Append("--pass");
+                Builder.AppendQuotedSecret(Settings.Password);
+            }
+
+            AppendArgumentIfNotNull("configFile", Settings.ConfigurationFile);
+
+            if (Settings.EnableDebugLogging)
+            {
+                Builder.Append("--debug");
+            }
+
+            if (Settings.IgnoreSslErrors)
+            {
+                Builder.Append("--ignoreSslErrors");
+            }
+
+            if (Settings.EnableServiceMessages)
+            {
+                Builder.Append("--enableServiceMessages");
+            }
+        }
+    }
+}
```

Comment:
@agc93 there is an extra line at the end of this file which can be removed :smile: